libidset: idset.h should include <stdbool.h>

When including `<flux/idset.h>` alone,  the compiler throws an error that type name `'bool'` is undefined

```
/usr/include/flux/idset.h:70:1: error: unknown type name 'bool'
 bool idset_test (const struct idset *idset, unsigned int id);
 ^
```

Since libidset can be used standalone it should include `<stdbool.h>`

--- a/src/common/libidset/idset.h
+++ b/src/common/libidset/idset.h
@@ -17,6 +17,7 @@
 
 #include <sys/types.h>
 #include <sys/param.h>
+#include <stdbool.h>
 
 enum idset_flags {
     IDSET_FLAG_AUTOGROW = 1, // allow idset size to automatically grow

